### ISSUE TITLE
Opening help files according to `help-buffer-options`

### ISSUE THREAD

**Issue by simonmandlik** [catgoose/telescope-helpgrep.nvim#6]
I know this is in `TODO`s in `README.md`, but for anybody wanting to open help files with this extension in a way that is more aligned with the standard `:h` (e.g. setting the buffer as nonmodifiable), this is how I do it:

```lua

local function helpgrep_mappings()
    local actions = require "telescope.actions"
    local function set_help_options(action)
        return function(...)
            action(...)
            -- see :h help-buffer-options
            vim.opt_local.foldmethod = "manual"
            vim.opt_local.tabstop = 8
            vim.opt_local.arabic = false
            vim.opt_local.binary = false
            vim.opt_local.buflisted = false
            vim.opt_local.cursorbind = false
            vim.opt_local.diff = false
            vim.opt_local.foldenable = false
            vim.opt_local.list = false
            vim.opt_local.modifiable = false
            vim.opt_local.number = false
            vim.opt_local.relativenumber = false
            vim.opt_local.rightleft = false
            vim.opt_local.scrollbind = false
            vim.opt_local.spell = false
        end
    end

    local mappings = {
        ["<CR>"] = set_help_options(actions.select_default),
        ["<C-s>"] = set_help_options(actions.select_horizontal),
        ["<C-v>"] = set_help_options(actions.select_vertical),
        ["<C-t>"] = set_help_options(actions.select_tab),
    }
    return { i = mappings, n = mappings }
end
```

and then

```lua
telescope.setup({
  ...
  extensions = {
    helpgrep = {
      mappings = helpgrep_mappings()
    }
  }
})
require("telescope").load_extension("helpgrep")
```

**Comment by catgoose** [comment]
Maybe we can make a setup opt to apply these to help files.

**Comment by simonmandlik** [comment]
Yes, I can prepare a PR to make `helpgrep_mappings` available in the extension, your call :)

I also thought about using `help`, `vert help`  and `tab help` and use these built-in commands to open the result, which would simplify things a lot, but AFAIK these commands work only for help tags and there is no straightforward way to open help files at exact lines.

**Comment by simonmandlik** [comment]
Thinking about it, `help` can open files directly, so perhaps `help <result_file>` followed by `vim.api.nvim_win_set_cursor` would work?

**Comment by catgoose** [comment]
> Yes, I can prepare a PR to make `helpgrep_mappings` available in the extension, your call :)
> 
> I also thought about using `help`, `vert help` and `tab help` and use these built-in commands to open the result, which would simplify things a lot, but AFAIK these commands work only for help tags and there is no straightforward way to open help files at exact lines.

How about `local_opts` that can either accept a boolean or table of opts?  True would indicate your defaults.

**Comment by catgoose** [comment]
> Thinking about it, `help` can open files directly, so perhaps `help <result_file` followed by `vim.api.nvim_win_set_cursor` would work?

Yeah I have had this thought before, but then I realize the problem that this plugin solves is trying to search help files for which have no associated help tags.  You can use `:Telescope help_tags` to search through those.

`:Telescope help_tags` search for `backup`:

![image](https://github.com/user-attachments/assets/9a5112f7-5f17-48a3-82a6-3ff71152ccc2)


`:Telescope help_grep` search for `backup`:

![image](https://github.com/user-attachments/assets/a0a20e45-0daa-422d-a4eb-398f971fdbda)

Also when this plugin opens a file, I don't think you can navigate between tags using `C-]` (I think that's it), because rather than open `:help foo`, it opens the `foo` help file in a new buffer.

**Comment by simonmandlik** [comment]
> Yeah I have had this thought before, but then I realize the problem that this plugin solves is trying to search help files for which have no associated help tags. 

Sure.

The ideal functionality would be to open `help` exactly as `:help` does but open not at a help tag location but pretty much anywhere where the match is.

It is possible to get filename and line number from telescope. 

Say I'm searching for this

<img width="1212" alt="image" src="https://github.com/user-attachments/assets/e11cc553-06d5-4abb-8d19-12921eb1d846" />

I can open it as a standard buffer (which this plugin currently does). But I can also do the following

```lua
-- both can be obtained from telescope
local help_file = "options.txt"
local pos = { 5962, 8 }
vim.cmd("help " .. help_file)
vim.api.nvim_win_set_cursor(0, pos)
```

which gets me to the exact location:

<img width="670" alt="image" src="https://github.com/user-attachments/assets/338d286f-6730-41e9-a4a9-b66e72995582" />

I think this is much more simple and easier to maintain than emulating all help file options. What do you think?

**Comment by catgoose** [comment]
Nice, I like it.  If you want to draft a PR  that would be great.

I've been busy because of the holidays so if I don't respond right away I'll get to it I promise.

**Comment by simonmandlik** [comment]
Opened here https://github.com/catgoose/telescope-helpgrep.nvim/pull/7

**Comment by catgoose** [comment]
Thanks!  I'll take a look at this today or tomorrow

**Comment by simonmandlik** [comment]
@catgoose did you manage to take a look? thanks :)

**Comment by catgoose** [comment]
Been busy with the holidays.  I made a comment on your PR!

**Comment by catgoose** [comment]
Closed via #7
